_sc2data/Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ValidatorData.xml

```diff
@@ -320,6 +342,409 @@
         <CombineArray value="AP_IsVanillaCommandCenterFlying"/>
         <CombineArray value="AP_IsAPCommandCenterFlying"/>
     </CValidatorCombine>
+    <CValidatorCombine id="AP_IsInvulnerableAndHasSOAHeroicShieldImmunity">
+        <Type value="And"/>
+        <CombineArray value="AP_IsInvulnerable"/>
+        <CombineArray value="AP_HasSOAHeroicShieldImmunity"/>
+    </CValidatorCombine>
+    <CValidatorUnitType id="AP_IsSOACaster">
+        <Value value="AP_SoACaster"/>
+    </CValidatorUnitType>
+    <CValidatorUnitType id="AP_IsVanillaSOACaster">
+        <Value value="SoACaster"/>
+    </CValidatorUnitType>
+    <CValidatorUnitFilters id="AP_IsInvulnerable">
+        <Filters value="Invulnerable;-"/>
+    </CValidatorUnitFilters>
+    <CValidatorUnitType id="AP_IsAPHerculesCombine"/>
+    <CValidatorCombine id="AP_IsVanillaHerculesCombine">
+        <CombineArray value="AP_IsVanillaHercules"/>
+        <CombineArray value="AP_IsVanillaHerculesLander"/>
+        <CombineArray value="AP_IsVanillaHerculesLanderFlying"/>
+    </CValidatorCombine>
+    <CValidatorCombine id="IsHerculesCombine" parent="CValidatorCombine">
+        <!-- Override -->
+        <CombineArray value="AP_IsVanillaHerculesCombine"/>
+        <CombineArray value="AP_IsAPHerculesCombine"/>
+    </CValidatorCombine>

```

Comment:
You can't do this as `IsHerculesCombine` exists in `void.sc2Campaign` but not `Liberty.SC2Mod`

You can apply this pattern only on `Liberty.SC2Mod` Validators. If you have a void map where two mutually unmergable validators exist, weird stuff will happen

Reply:
The manual says so:
```
NOTE: Apply this pattern if the validator is part of Liberty.SC2Mod, otherwise do the plain duplicate (AP_IsSomeStuff). Merging then happens at campaign-specific mod. See Vulture
```